src/librustc/middle/traits/error_reporting.rs

```diff
@@ -290,7 +290,7 @@ pub fn maybe_report_ambiguity<'a, 'tcx>(infcx: &InferCtxt<'a, 'tcx>,
                     {
                         span_err!(infcx.tcx.sess, obligation.cause.span, E0282,
                                 "unable to infer enough type information about `{}`; \
-                                 type annotations required",
+                                 type annotations or generic parameter binding required",
```

Comment:
maybe I misinterpreted the log files, but it certainly seemed like the `compile-fail` tests did need updating.  If you did not see failures on a windows `make check`, then that is quite troubling...